site/docs/expressions/scalar_functions.md

```diff
@@ -1,52 +1,118 @@
 
-Substrait supports a number of functions. A scalar function signature includes the following properties:
+| Property               | Description                                                  | Required                            |
+| ---------------------- | ------------------------------------------------------------ | ----------------------------------- |
+| Name                   | One or more user friendly utf8 strings that are used to reference this function in languages | At least one value is required.     |
+| List of arguments      | Argument properties are defined below. Arguments can be fully defined, parameterized or wildcarded. See further details below. | Optional, defaults to niladic.      |
+| Deterministic          | Whether this function is expected to reproduce the same output when it is invoked multiple times with the same input. This informs a plan consumer on whether it can constant reduce the defined function. An example would be a random() function, which is typically expected to be evaluated repeatedly despite having the same set of inputs. | Optional, defaults to true.         |
+| Session Dependent      | Whether this function is influenced by the session context it is invoked within. For example, a function may be influenced by a user who is invoking the function, the time zone of a session, or some other non-obvious parameter. This can inform caching systems on whether a particular function is cacheable. | Optional, defaults to false.        |
+| Variadic Behavior      | Whether the last argument of the function is variadic. Options include: [Single argument, M..N]. M must be less than or equal to N. N can be defined or left as "unlimited". | Optional, defaults to single value. |
+| Description            | Additional description of function for implementers or users. Should be written human readable to allow exposure to end users. Presented as a map with language => description mappings. E.g. `{ "en": "This adds two numbers together.", "fr": "cela ajoute deux nombres"}`. | Optional                            |
+| Return Type Expression | The output type is expected to be a physical type of the expression. Return types can be one of a simple, fully-defined compound type or a type expression. See below for more on type expressions. | Required                            |
+| Implementation Map     | A map of implementation locations for one or more implementations of the given function. Each key is a function implementation type. Implementation types include examples such as: AthenaArrowLambda, TrinoV361Jar, ArrowCppKernelEnum, GandivaEnum, LinkedIn Transport Jar, etc. [Definition TBD]. Implementation type has one or more properties associated with retrieval of that implementation. | Optional                            |
```

Comment:
Maybe a foolish ask but is there any chance we can get away with a URI here?

Reply:
My thinking was that providing the type of implementation in the map, it allows a system to only read manifests for implementations it can actually use. E.g. type => git-uri.